cmd/deploy/input_test.go

```diff
@@ -2,9 +2,240 @@ package main
 
 import (
 	"encoding/json"
+	"reflect"
 	"testing"
+
+	"cuelabs.dev/go/oci/ociregistry"
+	"github.com/docker-library/meta-scripts/registry"
 )
 
+func TestGetRefs(t *testing.T) {
+	type want struct {
+		digest ociregistry.Digest
+		refs   []registry.Reference
+	}
+	type fixture struct {
+		name       string
+		rawDigests []string
+		deployType deployType
+		want       want
+		wantErr    string
+	}
+	for _, x := range []fixture{
+		{
+			"happy path",
+			[]string{
+				"localhost:5000/foo@sha256:9ef42f1d602fb423fad935aac1caa0cfdbce1ad7edce64d080a4eb7b13f7cd9d",
+				"localhost:5000/bar:some-tag",
+				"localhost:5000/baz",
+			},
+			typeManifest,
+			want{
+				"sha256:9ef42f1d602fb423fad935aac1caa0cfdbce1ad7edce64d080a4eb7b13f7cd9d",
+				[]registry.Reference{
+					{
+						Host:       "localhost:5000",
+						Repository: "foo",
+						Digest:     "sha256:9ef42f1d602fb423fad935aac1caa0cfdbce1ad7edce64d080a4eb7b13f7cd9d",
+					},
+					{
+						Host:       "localhost:5000",
+						Repository: "bar",
+						Tag:        "some-tag",
+					},
+					{
+						Host:       "localhost:5000",
+						Repository: "baz",
+					},
+				},
+			},
+			"",
+		},
+		{
+			"invalid digest",
+			[]string{
+				"localhost:5000/foo@bad_digest",
+			},
+			typeManifest,
+			want{},
+			"localhost:5000/foo@bad_digest: failed to parse ref: invalid digest \"bad_digest\": invalid checksum digest format",
+		},
+		{
+			"invalid host",
+			[]string{
+				"...",
+			},
+			typeManifest,
+			want{},
+			"...: failed to parse ref: invalid reference syntax (\"...\")",
+		},
+		{
+			"invalid tag",
+			[]string{
+				"localhost:5000/foo:#",
+			},
+			typeManifest,
+			want{},
+			"localhost:5000/foo:#: failed to parse ref: tag \"#\" does not start with word character",
+		},
+		{
+			"invalid reference url",
+			[]string{
+				"localhost:5000/foo?test",
+			},
+			typeManifest,
+			want{},
+			"localhost:5000/foo?test: failed to parse ref: tag \"5000/foo?test\" contains invalid invalid character '/'",
```

Comment:
These are great examples of why testing the exact error message makes me nervous -- if Go decides (for whatever reason) to pull in a newer version of the `ociregistry` module, these error messages might all change because they ultimately come from that code, so we're effectively testing _their_ code instead of ours.  I don't know that there's a better answer here except maybe _only_ looking for `failed to parse ref` (our bit of the error) in the error message? (https://pkg.go.dev/strings#Contains)